python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,576 @@
+#
+# LSST Data Management System
+#
+# Copyright 2008-2015  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from contextlib import closing
+import MySQLdb, MySQLdb.constants, MySQLdb.converters
+import math
+import re
+import struct
+
+import lsst.afw.table as afwTable
+from lsst.daf.persistence import DbAuth
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+## \addtogroup LSST_task_documentation
+## \{
+## \page daf_ingest_IngestCatalogTask
+## \ref IngestCatalogTask_ "IngestCatalogTask"
+##      \copybrief IngestCatalogTask
+## \}
+
+class FieldFormatter(object):
+    """A class for formatting fields in an afw BaseCatalog.
+
+    This class is a container for a function that maps a Field to a MySQL type,
+    and a function that maps a field value to a literal suitable for use in a
+    MySQL INSERT/REPLACE statement.
+    """
+
+    def __init__(self, sqlTypeCallable, formatValueCallable):
+        """Store the field formatting information.
+        """
+        self.sqlTypeCallable = sqlTypeCallable
+        self.formatValueCallable = formatValueCallable
+
+    def sqlType(self, field):
+        """Return the SQL type for the given field's values.
+        """
+        return self.sqlTypeCallable(field)
+
+    def formatValue(self, value):
+        """Return a string representation of value suitable for use in an
+        INSERT/REPLACE statement (not a CSV file).  Nones are always
+        transformed to SQL NULLs.
+        """
+        if value is None:
+            return "NULL"
+        return self.formatValueCallable(value)
+
+
+def _formatNumber(formatString, number):
+    """Helper function for formatting floating point numbers. NaNs and
```

Comment:
It is not the same - formatValue is for any python type, _formatNumber is a function for formatting floating point numbers specifically. I'm mapping NaN and +-Inf to NULL, because MySQL does not support storing them in FLOAT/DOUBLE columns. I'll update the comment to say that explicitly.